vendor/bat-native-ledger/src/bat/ledger/internal/core/bat_ledger_context.cc

```diff
@@ -7,44 +7,63 @@
 
 #include <utility>
 
+#include "bat/ledger/internal/constants.h"
 #include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/ledger.h"
 #include "bat/ledger/ledger_client.h"
+#include "bat/ledger/option_keys.h"
+#include "bat/ledger/public/interfaces/ledger.mojom.h"
 
 namespace ledger {
 
 namespace {
 
-size_t g_next_component_key = 0;
+BATLedgerContext::Environment GetEnvironment() {
+  switch (ledger::_environment) {
+    case mojom::Environment::DEVELOPMENT:
+      return BATLedgerContext::Environment::kDevelopment;
+    case mojom::Environment::STAGING:
+      return BATLedgerContext::Environment::kStaging;
+    case mojom::Environment::PRODUCTION:
+      return BATLedgerContext::Environment::kProduction;
+  }
+}
+
+BATLedgerContext::Options GetOptions(LedgerClient* ledger_client) {
+  DCHECK(ledger_client);
```

Comment:
I think this is fine as-is for now; but if more items are added to options (specifically strings or pointers), there could be a problem as the return value is going to get copied (implicit copy constructor)

What would you think about changing to a `SetOptions` and then calling inside body of constructor for `BATLedgerContext`?